app/src/main/java/com/yacgroup/yacguide/network/SectorParser.kt

```diff
@@ -164,6 +170,7 @@ class SectorParser(private val _db: DatabaseWrapper,
             r.parentId = sectorId
             _rocks.add(r)
         }
+        _listener.onUpdateStatus("${ (100 * (++_parsedSectorCount)) / _sectorCount } %")
```

Comment:
Is it guaranteed, that `_sectorCount` is always greater than zero?

Reply:
Good point! However, if `_sectorCount` is 0, no rocks will be requested and hence, this method will never be executed.